src/herad.cpp

```diff
@@ -69,9 +69,9 @@ std::string CheradPlayer::gettype()
 {
 	char scomp[12 + 1] = "";
 	if (comp > HERAD_COMP_NONE)
-		sprintf(scomp, ", %s packed", (comp == HERAD_COMP_HSQ ? "HSQ" : "SQX"));
-	char type[40 + 1];
-	sprintf(type, "HERAD System %s (version %d%s)", (AGD ? "AGD" : "SDB"), (v2 ? 2 : 1), scomp);
+		snprintf(scomp, sizeof(scomp), ", %s packed", (comp == HERAD_COMP_HSQ ? "HSQ" : "SQX"));
+	char type[41 + 1];
```

Comment:
40 + 1 is already the size with terminating zero, no need to change that.

Reply:
```
 1234567890123456789012345678901234567890
"HERAD System AGD (version 2, HSQ packed)"
```
Yes you are right, 40 + 1 is enough.